src/sentry/static/sentry/app/views/settings/incidentRules/metricField.tsx

```diff
@@ -0,0 +1,200 @@
+    validDataset: [Dataset.TRANSACTIONS],
+    default: 'count()',
+  },
+  {
+    match: /^error_rate\(\)/,
+    name: 'Error rate',
+    validDataset: [Dataset.TRANSACTIONS],
+    default: 'error_rate()',
+  },
+];
+
+type OptionConfig = {
+  aggregations: AggregationKey[];
+  fields: FieldKey[];
+};
+
+const errorFieldConfig: OptionConfig = {
+  aggregations: ['count', 'count_unique'],
+  fields: ['user'],
+};
+
+const transactionFieldConfig: OptionConfig = {
+  aggregations: [
+    'avg',
+    'percentile',
+    'error_rate',
+    'apdex',
+    'count',
+    'p50',
+    'p75',
+    'p95',
+    'p99',
+    'p100',
+  ],
+  fields: ['transaction.duration'],
+};
+
+const getFieldOptionConfig = (dataset: Dataset) => {
+  const config = dataset === Dataset.ERRORS ? errorFieldConfig : transactionFieldConfig;
+
+  const aggregations = Object.fromEntries(
+    config.aggregations.map(key => [key, AGGREGATIONS[key]])
+  );
+  const fields = Object.fromEntries(config.fields.map(key => [key, FIELDS[key]]));
+
+  return {aggregations, fields};
+};
+
+const help = ({name, model}: {name: string; model: FormModel}) => {
+  const aggregate = model.getValue(name) as string;
+
+  const presets = cannedAggregates
+    .filter(preset => preset.validDataset.includes(model.getValue('dataset') as Dataset))
+    .map(preset => ({...preset, selected: preset.match.test(aggregate)}))
+    .map((preset, i, list) => (
+      <React.Fragment key={preset.name}>
+        <Tooltip title={t(`This preset is selected`)} disabled={!preset.selected}>
+          <PresetLink
+            onClick={() => model.setValue(name, preset.default)}
+            isSelected={preset.selected}
+          >
+            {preset.name}
+          </PresetLink>
+        </Tooltip>
+        {i + 1 < list.length && ', '}
+      </React.Fragment>
+    ));
+
+  return tct(
+    'Choose an aggregate function. Not sure what to select, try a preset: [presets]',
+    {presets}
+  );
+};
+
+const MetricField = ({organization, ...props}: Props) => (
+  <FormField help={help} {...props}>
+    {({onChange, value, model}) => {
+      const dataset = model.getValue('dataset');
+
+      const fieldOptionsConfig = getFieldOptionConfig(dataset);
+      const fieldOptions = generateFieldOptions({organization, ...fieldOptionsConfig});
+      const fieldValue = explodeFieldString(value ?? '');
+
+      const fieldKey =
+        fieldValue?.kind === FieldValueKind.FUNCTION
+          ? `function:${fieldValue.function[0]}`
+          : '';
+
+      const selectedField = fieldOptions[fieldKey]?.value;
+      const numParameters =
+        selectedField &&
+        selectedField.kind === FieldValueKind.FUNCTION &&
+        selectedField.meta.parameters.length;
+
+      return (
+        <React.Fragment>
+          <AggregateHeader>
+            <div>{t('Function')}</div>
+            {numParameters > 0 && <div>{t('Parameter')}</div>}
+            {numParameters > 1 && <div>{t('Refinement')}</div>}
```

Comment:
@markstory There was no header for this in the builder. But I think this makes sense, maybe we should replicate over there?

Reply:
I left out the header before as I was concerned with exposing more terminology to the user. They don't generally need to know that we call it 'refinement' internally, it is just another parameter to them.

Reply:
i'll drop it too 👍